dice-game/service/src/main.rs

```diff
@@ -0,0 +1,153 @@
+use map_vec::Map;
+use oasis_std::{Address, Context};
+use rand::Rng;
+
+#[derive(oasis_std::Service)]
+struct DiceGame {
+    max_score: u32,
+    num_players: u32,
+    scores: Map<Address, u32>,
+}
+
+type Result<T> = std::result::Result<T, String>;
+impl DiceGame {
+    pub fn new(_ctx: &Context, num_players: u32) -> Self {
+        Self {
```

Comment:
```suggestion
        Default::default()
```

Reply:
I'm passing in num_players into the constructor so I don't think `Default::default()` gives me the behavior needed here